include/mqt-core/na/operations/NAShuttlingOperation.hpp

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include "na/NADefinitions.hpp"
+#include "na/operations/NAOperation.hpp"
+#include "operations/OpType.hpp"
+
+#include <cmath>
+#include <utility>
+#include <vector>
+namespace na {
+
+enum ShuttleType : std::uint8_t { LOAD, MOVE, STORE };
+
+class NAShuttlingOperation : public NAOperation {
+protected:
+  ShuttleType type;
+  std::vector<std::shared_ptr<Point>> start;
+  std::vector<std::shared_ptr<Point>> end;
```

Comment:
Similar to the local operation class, these can just be regular `Point`s instead of shared pointers.